### PR TITLE
feat: enable GIF preview and add Action Buttons

### LINKED ISSUE #15: [feat] Enable user to view GIF preview and provide options to regenerate/approve the generation
## Issue Description

Currently, in the `/gen-gif q "query"` command the GIF is directly sent to the channel, however we should provide user a way to preview the generation before it is sent to the channel. To achieve this we can by creating a self view message with the GIF and the `Fuselage UIKit` to implement a bar of buttons for regenerate/approve.

### PR DESCRIPTION
**Pull request by AmanNegi**
## Description :memo: 

- Implement UIKit components and Buttons to provide user with an option to preview the GIF before sending to the channel.
- The user can decide to re-generate the GIF or send the generated GIF to the channel.

> Fixes: #15 

## Demo :video_camera: 

https://github.com/user-attachments/assets/e491ddbb-3263-4c6f-8e3a-4cb0b88462c7

### CODE REVIEW

**Comment by AmanNegi** [comment]
@Nabhag8848 Could you please review this and let me know if any changes are required.

**Comment by Nabhag8848** [comment]
@AmanNegi  merge conflicts

**Comment by AmanNegi** [comment]
> @AmanNegi merge conflicts

Fixed the merge conflicts. :heavy_check_mark:

File: src/handlers/ExecuteBlockActionHandler.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        if (blockId === ButtonBlockIds.REGENERATE_OPTIONS_BLOCK) {
+            if (actionId === ButtonActionIds.APPROVE && value && room) {
+                await this.sendGifToRoom({
+                    modify: this.modify,
+                    room,
+                    user,
+                    prompt: args.prompt,
+                    url: args.url,
+                });
+            } else if (
```
Comment (Nabhag8848) [inline src/handlers/ExecuteBlockActionHandler.ts]: @AmanNegi use switch instead.

File: src/lib/RedefinePrompt.ts [hunk 2]
```diff
@@ -60,6 +60,8 @@ export class RedefinedPrompt {
                 throw new Error("Failed to get response from model");
             }
 
+            logger.log("PromptVariationCommand.preview", response.data);
```
Comment (Nabhag8848) [inline src/lib/RedefinePrompt.ts]: @AmanNegi

File: src/handlers/PreviewerHandler.ts [hunk 3]
```diff
@@ -115,14 +116,17 @@ export class PreviewerHandler {
...
+
+        if (res) {
+            items = res.map((item) => {
+                return {
+                    id: item.prompt,
+                    type: SlashCommandPreviewItemType.TEXT,
+                    value: item.prompt,
+                };
+            });
+        }
```
Comment (Nabhag8848) [inline src/handlers/PreviewerHandler.ts:137]: @AmanNegi What if the res is not there how user will get to know ? as discussed.

**Review by Nabhag8848 — CHANGES_REQUESTED** [review]
@AmanNegi

**Comment by AmanNegi** [comment]
@Nabhag8848 I have made the requested changes. Could you please take a look!

**Review by Nabhag8848 — APPROVED** [review]
@AmanNegi LGTM !

**Pull request merged**